### PR TITLE
Merge/sprint 1

### PR DESCRIPTION
**Pull request by AsafKaravani**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -420,6 +439,28 @@
...
+                <img class="audioImg" src="icons/voiceMessage/group_5.png"/>
+               <div class="audio">
+                    <audio id="audioSRC" src="icons/voiceMessage/file_example_MP3_700KB.mp3" ontimeupdate="updateAudioMessageTime()"></audio>
+                    <span id="audioTime" class="audioTime">00:00</span>
+                    <img id="audioTrack" class="audioTrack" src="icons/voiceMessage/rectangle.png"/>
+                    <img id="trackerPoint" class="trackerPoint" src="icons/voiceMessage/blue.png"/>
+                    <img class="audioPlay" onclick="playBTN()" src="icons/voiceMessage/triangle.png"/>
+                </div>
+                <div class="border"></div>
+                <p class="wroteMessage">שלום, כאן א', מפקד טייסת אבירי הזנב הכתום. אנו אנשי הטייסת, הטכנאים והלוחמים בסדיר ובמילואים, מבקשים לחזק את ידם של הצוותים הרפואיים וכלל האזרחים בימים לא פשוטים אלו. מאחלים לכולנו חג עצמאות שמח! </p>
```
Comment (leo212) [inline index.html:456]: This text should be dynamic, using information from json files.

File: index.html [hunk 2]
```diff
@@ -420,6 +439,28 @@
...
+            </div>
+            <div class="sendPost">
+                <img class="camera" src="icons/voiceMessage/group_4.png"/>
+                <span class="sendMessage"><p>העלו תמונה מהמטס והשתתפו בתחרות!</p></span>
```
Comment (leo212) [inline index.html:460]: What is this? How can the use upload photos?

File: js/functions.js [hunk 3]
```diff
@@ -2211,3 +2213,109 @@ function getEventDescription(isAerobatics, locationName, minutes) {
...
+
+
+(function () {
+    var userLoc = null;
+    navigator.geolocation.watchPosition(function(newLoc){
+        userLoc = newLoc;
+        userLoc = {lon: userLoc.coords.longitude, lat: userLoc.coords.latitude};
+    });
+
+    var aircraftsName = [
```
Comment (leo212) [inline js/functions.js:2226]: This info already loaded from  aircraftinfo.json file, there is no need for that conversation table. Take it from the aircraft object itself.

File: js/functions.js [hunk 4]
```diff
@@ -2211,3 +2213,109 @@ function getEventDescription(isAerobatics, locationName, minutes) {
...
+        {'hebName':'יסעור','engName':'yasur'}
+    ];
+
+    function notifyUserIfNear(currentLocation, aircraft) {
+        if (userLoc) {
+            currentLocation = {lon: currentLocation.lng, lat: currentLocation.lat};
+            if(haversineDistance(userLoc,currentLocation) < 1000)
+            {
+                var name = aircraft.name;
+                document.getElementById("gottoVoiceMessagePopup").style.display = "block";
```
Comment (leo212) [inline js/functions.js:2262]: A general comment for all of document.getElementById - it is preferred to use jQuery syntax instead to find object hide or show them.

File: js/functions.js [hunk 5]
```diff
@@ -2211,3 +2213,109 @@ function getEventDescription(isAerobatics, locationName, minutes) {
...
+                document.getElementById("aircraftName").innerHTML = " מטוס קרב " + name;
```
Comment (leo212) [inline js/functions.js:2263]: Use backquote syntax instead of string concatination.

File: js/map.js [hunk 6]
```diff
@@ -410,8 +410,8 @@ googleMaps = {
     createMapObject: (clickCallback) => {
         map = new google.maps.Map(document.getElementById('map'),
             {
-                center: {lat: 32.00, lng: 35.00},
-                zoom: 8,
+                center: {lat: 31.20, lng: 34.97},
+                zoom: ($(window).height() > 950) ? 8 : 7,
```
Comment (leo212) [inline js/map.js:414]: why? almost any device now has more then 950 pixels height.

File: js/popups.js [hunk 7]
```diff
@@ -825,9 +830,63 @@ function closeMapClusterPopup(clearMap) {
...
+        }
+        else {
+            audioPLay = true;
+            document.getElementById("audioSRC").play();
+        }
+    }
+}
+
+function updateAudioMessageTime() {
+    activeVoiceMessage = document.getElementById("audioSRC");
```
Comment (leo212) [inline js/popups.js:882]: use jquery instead of document.getElementById

File: js/popups.js [hunk 8]
```diff
@@ -825,9 +830,63 @@ function closeMapClusterPopup(clearMap) {
...
+function showAudioMessagePopup() {
+    document.getElementById("gottoVoiceMessagePopup").style.display = "none";
+    document.getElementById("myModal").style.display = "flex";
+    playAudioMessageAndTracker()
+}
+
+function closeVoiceMessagePopup(){
+    document.getElementById("audioSRC").pause();
+    document.getElementById("audioSRC").currentTime = 0;
+    document.getElementById("myModal").style.display = "none";
```
Comment (leo212) [inline js/popups.js:854]: use $("#myModal").hide() instead.

File: js/popups.js [hunk 9]
```diff
@@ -825,9 +830,63 @@ function closeMapClusterPopup(clearMap) {
...
+    //closeVoiceMessagePopup();
     closeMapClusterPopup(true);
 }
 
 $(document).ready(function () {
     //window.scrollTo(0,document.body.scrollHeight);
 });
+
+function closeGotoVoiceMessagePopup() {
+    document.getElementById("gottoVoiceMessagePopup").style.display = "none";
```
Comment (leo212) [inline js/popups.js:842]: use $("#gottoVoiceMessagePopup").hide() instead.

File: js/functions.js [hunk 10]
```diff
@@ -2211,3 +2218,134 @@ function getEventDescription(isAerobatics, locationName, minutes) {
...
+                        }
+                    }
+                //}
+            }
+
+            else if(haversineDistance(userLoc,currentLocation) > 2000)
+            {
+                if(document.getElementById("myModal").style.display !== "block")
+                {
+                    document.getElementById("gottoVoiceMessagePopup").style.display = "none";
```
Comment (leo212) [inline js/functions.js:2316]: use $("#gottoVoiceMessagePopup").hide() instead.

File: js/functions.js [hunk 11]
```diff
@@ -2211,3 +2218,134 @@ function getEventDescription(isAerobatics, locationName, minutes) {
...
+                        else{
+                            document.getElementById("aircraftImg").src="";
+                        }
+                    }
+                //}
+            }
+
+            else if(haversineDistance(userLoc,currentLocation) > 2000)
+            {
+                if(document.getElementById("myModal").style.display !== "block")
```
Comment (leo212) [inline js/functions.js:2314]: try to find cleaner way to check if it is visible... perhaps use a variable.

File: js/functions.js [hunk 12]
```diff
@@ -2211,3 +2218,134 @@ function getEventDescription(isAerobatics, locationName, minutes) {
...
+                        }
+                        if(audioSrc !== undefined && audioSrc !== ""){
+                            document.getElementById("audioSRC").src=audioSrc;
+                        }
+                        else{
+                            document.getElementById("audioSRC").src="";
+                        }
+
+                        if(engName !== undefined && engName !== ""){
+                            document.getElementById("aircraftImg").src="icons/aircrafts/"+engName+".png";
```
Comment (leo212) [inline js/functions.js:2303]: use ${"#aircraftImg").attr("src",....) instead

File: js/functions.js [hunk 13]
```diff
@@ -2211,3 +2218,134 @@ function getEventDescription(isAerobatics, locationName, minutes) {
...
+                    var x = setInterval( function () { timeCount++;}, 1000);
+                } */
+                //check that its not the same popup that has been closed - if its not the same aircraft or its has been more then 30 sec
+                //if(timeCount >30 || name !== aircraft.name) {
+                    if(document.getElementById("myModal").style.display === "none" && document.getElementById("gottoVoiceMessagePopup").style.display === "none")
+                    {
+                        //timeCount = 0;
+                        name = aircraft.name;
+                        document.getElementById("gottoVoiceMessagePopup").style.display = "block";
+                        document.getElementById("aircraftName").innerHTML = " מטוס קרב " + name;
```
Comment (leo212) [inline js/functions.js:2281]: use use ${"#aircraftImg").html() instead

Comment (leo212) [inline js/functions.js:2281]: it is not always "מטוס קרב", get its type from aircraft object.

File: index.html (hunk 2 above)
Comment (AsafKaravani) [reply index.html]: This will be a link to IAF's photo contest. It is an external site.

File: index.html (hunk 1 above)
Comment (AsafKaravani) [reply index.html:457]: @mayPeleg

File: index.html (hunk 2 above)
Comment (AsafKaravani) [reply index.html]: The will be available only a few days before the event.
For now we will link it to IAF's instagram.

Comment (AsafKaravani) [reply index.html]: @mayPeleg Please, link this ".sendPost" div to IAF instagram.

File: index.html (hunk 1 above)
Comment (AsafKaravani) [reply index.html:457]: @mayPeleg Please, add to the json a "message" field, and use it dynamically.

File: js/functions.js (hunk 3 above)
Comment (AsafKaravani) [reply js/functions.js:2230]: We will check that.
@mayPeleg Call me, lets figure it out.

File: js/functions.js (hunk 4 above)
Comment (AsafKaravani) [reply js/functions.js]: We will change all the getElementById occurrences. @mayPeleg

Comment (AsafKaravani) [reply js/functions.js]: Note: getElementById is almost 3 times faster  than jQuery.
https://www.measurethat.net/Benchmarks/Show/3/0/jquery-by-id-vs-documentgetelementbyid

File: js/functions.js (hunk 5 above)
Comment (AsafKaravani) [reply js/functions.js]: @mayPeleg Use template string.
Here is a link to what it is: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: js/popups.js (hunk 8 above)
Comment (AsafKaravani) [reply js/popups.js:854]: @mayPeleg

File: js/popups.js (hunk 9 above)
Comment (AsafKaravani) [reply js/popups.js:842]: @mayPeleg

File: js/functions.js (hunk 10 above)
Comment (AsafKaravani) [reply js/functions.js:2316]: @mayPeleg

File: js/functions.js (hunk 11 above)
Comment (AsafKaravani) [reply js/functions.js:2314]: I will check @AsafKaravani

File: js/functions.js (hunk 12 above)
Comment (AsafKaravani) [reply js/functions.js:2303]: @mayPeleg

File: js/functions.js (hunk 13 above)
Comment (AsafKaravani) [reply js/functions.js:2281]: @mayPeleg

Comment (AsafKaravani) [reply js/functions.js:2281]: @mayPeleg Please insert the plane type via string templates too.

File: index.html (hunk 1 above)
Comment (AsafKaravani) [reply index.html:457]: Done

File: index.html (hunk 2 above)
Comment (AsafKaravani) [reply index.html]: Done

File: js/functions.js (hunk 4 above)
Comment (AsafKaravani) [reply js/functions.js]: Done

File: js/popups.js (hunk 7 above)
Comment (AsafKaravani) [reply js/popups.js]: Done

File: js/functions.js (hunk 5 above)
Comment (AsafKaravani) [reply js/functions.js]: Done

File: js/popups.js (hunk 8 above)
Comment (AsafKaravani) [reply js/popups.js]: Done

File: js/functions.js (hunk 10 above)
Comment (AsafKaravani) [reply js/functions.js:2327]: Done

File: js/popups.js (hunk 9 above)
Comment (AsafKaravani) [reply js/popups.js]: Done

File: js/functions.js (hunk 12 above)
Comment (AsafKaravani) [reply js/functions.js]: Done

File: js/functions.js (hunk 13 above)
Comment (AsafKaravani) [reply js/functions.js]: Done

**Pull request merged**